impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Gauge.js

```diff
@@ -0,0 +1,155 @@
+  "pentaho/visual/util"
+], function(module, BaseView, echarts, util) {
+
+  "use strict";
+
+  return BaseView.extend(module.id, {
+
+    _buildSeries: function() {
+      var dataTable = this.model.data;
+      var range = dataTable.getColumnRange(dataTable.getNumberOfColumns() - 1);
+      var rangeLength = range.max.toString().length;
+      var div = Math.pow(10, rangeLength - 1);
+
+      return [
+        {
+          type: "gauge",
+          min: 0,
+          max: Math.ceil(range.max / div) * div,
+
+          pointer: {
+            width: 8,
+            length: "80%",
+            offsetCenter: [0, "8%"]
+          },
+          progress: {
+            show: true,
+            overlap: true,
+            roundCap: true
+          },
+          detail: {
+            width: "auto",
+            height: "auto",
+            legendHoverLink: true,
+            fontSize: 14,
+            color: "#fff",
+            backgroundColor: "inherit",
+            borderRadius: 3,
+            formatter: "{value}"
+          },
+          itemStyle: {
+            borderColor: "#fff",
+            borderWidth: 1
+          },
+          emphasis: {
+            label: {
+              fontSize: 20
+            }
+          }
+        }
+      ];
+    },
+
+    _buildData: function() {
+      var dataTable = this.model.data;
+
+      if(dataTable.originalCrossTable) {
+        dataTable = dataTable.originalCrossTable.toPlainTable({skipRowsWithAllNullMeasures: true});
+      }
+
+      var rowLength = dataTable.getNumberOfRows();
+      var colLength = dataTable.getNumberOfColumns();
+      var font = util.getDefaultFont(null, 12);
+
+      var records = [];
+
+      var maxWidth = this.model.width;
+      var midIdx = Math.floor(rowLength / 2); // Calculate the index of the middle element
+
+      var initX = 0; // Midpoint x-coordinate
+      var yPercentage = 100; // Constant y-coordinate in percentage
+      var separation = (maxWidth / (rowLength - 1)) * 0.8; // Dynamic separation between elements
+
+      // Read data from dataTable and push it to records
+      for(var i = 0; i < rowLength; i++) {
+        var offsetX;
+
+        if(rowLength % 2 === 0) { // If even number of data elements
+          if(i < midIdx) { // Elements before the middle
+            offsetX = -((midIdx - 0.5 - i) * separation);
+          } else { // Elements after the middle
+            offsetX = (i - midIdx + 0.5) * separation;
+          }
+        } else { // If odd number of data elements
+          if(i === midIdx) { // Middle element
+            offsetX = 0;
+          } else if(i < midIdx) { // Elements before the middle
+            offsetX = -((midIdx - i) * separation);
+          } else { // Elements after the middle
+            offsetX = (i - midIdx) * separation;
+          }
+        }
+
+        records.push({
+          name: dataTable.getFormattedValue(i, colLength - 2),
+          value: dataTable.getValue(i, colLength - 1),
```

Comment:
This way of directly accessing columns by index, assuming a certain column layout is not correct. The data table column layout should not be assumed. Just noticed that the same was done on the Funnel chart view.

For example, the correct way to know the data table columns which were mapped to the `rows` and `measures` visual roles would be:
```js
var categoriesColIndexes = this.model.rows.fieldIndexes;
var measureColIndex = this.model.measures.fieldIndexes[0];
```